oximeter/collector/src/agent.rs

```diff
@@ -679,8 +679,10 @@ impl OximeterAgent {
         >,
         id: Uuid,
     ) -> Result<(), Error> {
-        let (_info, task) =
-            tasks.remove(&id).ok_or_else(|| Error::NoSuchProducer(id))?;
```

Comment:
I think this was the only use of `Error::NoSuchProducer`, so we could probably drop this variant entirely now?

Reply:
Yeah, good catch.

Reply:
Done in f4736972c